### DISCUSSION TITLE
Placeholder to Number mask with blocks

### DISCUSSION THREAD

**Discussion by sanyi9305** [uNmAnNeR/imaskjs#612] [Q&A] [answered]
Hi, 

Can I use placeholder with Number mask and blocks?
I would like number with placeholder:
example:
If I use a decimal number I would like show placeholder:  0.0000
![image](https://user-images.githubusercontent.com/32463572/152566078-abf14b19-c6b8-46c0-b458-90997ccb8f29.png)

This is my examaple code:
maskOptions = {
    mask: [{
            mask: positiveformat,
            startsWith: '+',
            placeholderChar: '#',
            lazy: false,
            blocks: {
                num: {
                    mask: Number,
                    lazy: false,
                    placeholderChar: '#',
                    overwrite: true,
                    scale: 2,
                    signed: signed,
                    thousandsSeparator: thousandsSeparator,
                    padFractionalZeros: padFractionalZeros,
                    normalizeZeros: normalizeZeros,
                    radix: radix,
                    mapToRadix: [radix]
                }
            }
        }, {
            mask: negativeFormat,
            startsWith: '-',
            lazy: false,
            blocks: {
                num: {
                    mask: Number,
                    lazy: false,
                    placeholderChar: '#',
                    overwrite: true,
                    scale: scale,
                    signed: signed,
                    thousandsSeparator: thousandsSeparator,
                    padFractionalZeros: padFractionalZeros,
                    normalizeZeros: normalizeZeros,
                    radix: radix,
                    mapToRadix: [radix]
                }
            }
        }, {
            mask: Number,
            startsWith: '',
            lazy: false,
            overwrite: true,
            scale: scale,
            signed: signed,
            thousandsSeparator: thousandsSeparator,
            padFractionalZeros: padFractionalZeros,
            normalizeZeros: normalizeZeros,
            radix: radix,
            mapToRadix: [radix],
            placeholderChar: '#'
        }
    ],

    dispatch: function (appended, dynamicMasked) {

        var number = (appended + dynamicMasked.value);

        return dynamicMasked.compiledMasks.find(function (m) {

            return number.indexOf(m.startsWith) === 0;

        });
    }

};

**Comment by uNmAnNeR** [comment, marked as answer]
hi! it's not possible number mask is fully arbitrary and it's length is not determined. So if you want placeholder like `0.000` or any other then there is no way to understand where is real value or it is a placeholder.
If you want to have a placeholder with number then you can use range mask. But for decimals it will be tricky.

**Reply by sanyi9305** [reply]
Thank you.
